src/core/store.ts

```diff
@@ -11,8 +11,9 @@ export class DefaultStore<T> implements HoneybadgerStore<T> {
         return this.store
     }
 
-    run(store: T, callback: (...args: never[]) => void, args: never): void {
-        this.store = store
-        callback(args)
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    run<R, TArgs extends any[]>(store: T, callback: (...args: TArgs) => R, ...args: TArgs): R {
```

Comment:
Maybe you could do:
```suggestion
    run<R, TArgs extends never[]>(store: T, callback: (...args: TArgs) => R, ...args: TArgs): R {
```
so that the eslint warning could go away?

Reply:
I initially had the args typed as `never[]` (in a different format). It worked until I had to reference the args (the return statement). Apparently, `never` is for when you don't need the args. But I'll try in this form.

Reply:
Seems like it worked!👍